lib/virtual-net/src/composite.rs

```diff
@@ -0,0 +1,121 @@
+use std::net::SocketAddr;
+use std::task::{Context, Poll};
+
+use crate::{Ipv4Addr, Ipv6Addr, NetworkError, VirtualIoSource, VirtualTcpListener};
+use derivative::Derivative;
+use virtual_mio::ArcInterestHandler;
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct CompositeTcpListener {
```

Comment:
In what context do we need this?
It's not used anywhere in the added code I think, just exposed?

Is it used in implementations?

Reply:
its used in `Edge` for the `DistributedNetworking`. It allows combining multiple listeners into a single listener which is useful when there are multiple switches (e.g. loopback networking and remote networking)